src/main/java/codesquad/codestagram/controller/UserController.java

```diff
@@ -1,60 +1,124 @@
+import codesquad.codestagram.service.UserService;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.mvc.support.RedirectAttributes;
 
 import java.util.List;
+import java.util.Objects;
 
 @Controller
 @RequestMapping("/users")
 public class UserController {
 
-    private final UserRepository userRepository;
+    private final UserService userService;
 
-    public UserController(UserRepository userRepository) {
-        this.userRepository = userRepository;
+    public UserController(UserService userService) {
+        this.userService = userService;
     }
 
     @PostMapping
-    public String signup(@ModelAttribute User user) {
-        userRepository.save(user);
+    public String signup(@ModelAttribute User user, RedirectAttributes redirectAttributes) {
+
+        try {
+            userService.createUser(user);
+        } catch (Exception e) {
+            redirectAttributes.addFlashAttribute("message", e.getMessage());
+            return "redirect:/users/form";
+        }
+
         return "redirect:/users";
     }
 
     @GetMapping
     public String viewUserList(Model model) {
-        List<User> users = userRepository.findAll();
+        List<User> users = userService.getUserList();
         model.addAttribute("users", users);
         return "user/list";
     }
 
     @GetMapping("/{id}")
-    public String viewUserProfile(@PathVariable("id") Long id, Model model) {
-        User user = userRepository.findById(id).orElseThrow();
-        model.addAttribute("user", user);
+    public String viewUserProfile(@PathVariable("id") Long id, Model model, RedirectAttributes redirectAttributes) {
+
+        try {
+            User user = userService.getUser(id);
+            model.addAttribute("user", user);
+        } catch (IllegalArgumentException e) {
+            redirectAttributes.addFlashAttribute("message", e.getMessage());
+            return "redirect:/";
+        }
         return "user/profile";
     }
 
     @GetMapping("/{id}/form")
-    public String viewUpdateForm(@PathVariable("id") Long id, Model model) {
-        User user = userRepository.findById(id).orElseThrow();
-        model.addAttribute("user", user);
+    public String viewUpdateForm(@PathVariable("id") Long id, Model model, HttpSession session
+        , RedirectAttributes redirectAttributes) {
+
+        if (session.getAttribute("loginUser") == null) { //로그인 상태인지 확인
+            return "redirect:/users/loginForm";
+        }
+
+        User loginUser = (User) session.getAttribute("loginUser");
+
+        try {
+            User findUser = userService.getUser(id);
+
+            if (loginUser.getId() != findUser.getId()) { //URL을 직접 입력해서 타인의 회원정보를 수정할 경우
+                throw new IllegalArgumentException("잘못된 접근입니다.");
```

Comment:
```suggestion
                throw new IllegalArgumentException("사용자 ID가 일치하지 않습니다.");
```
요즘은 구체적인 오류 메시지를 주는 걸 꺼리는 경우도 많지만 저는 구체적인 걸 선호합니다.